File: gma-next-gen-sdk-android-examples-main/java/nextgenexample/app/src/main/java/com/example/nextgenexample/nativead/CustomVideoControlsView.java
package com.example.nextgenexample.nativead;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import androidx.core.content.ContextCompat;
import com.example.nextgenexample.R;
import com.google.android.libraries.ads.mobile.sdk.common.VideoController;
import com.google.android.libraries.ads.mobile.sdk.nativead.MediaContent;

/**
 * This view represents the status of a video controller and also displays custom controls for the
 * video controller when appropriate.
 */
public class CustomVideoControlsView extends LinearLayout
    implements VideoController.VideoLifecycleCallbacks {
  private ImageButton playButton;
  private ImageButton muteButton;
  private View controlsView;
  private boolean isVideoPlaying;

  public CustomVideoControlsView(Context context) {
    super(context);
    init(context, null, 0);
  }

  public CustomVideoControlsView(Context context, AttributeSet attrs) {
    super(context, attrs);
    init(context, attrs, 0);
  }

  public CustomVideoControlsView(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    init(context, attrs, defStyle);
  }

  private void init(Context context, AttributeSet unusedAttrs, int unusedDefStyle) {
    LayoutInflater inflater =
        (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    inflater.inflate(R.layout.custom_video_controls, this, true);

    playButton = findViewById(R.id.btn_play);
    muteButton = findViewById(R.id.btn_mute);
    controlsView = findViewById(R.id.video_controls);
    controlsView.setVisibility(View.GONE);
  }

  /*
   * Sets up the custom controls view with the provided VideoController and mute state.
   */
  @SuppressLint("SetTextI18n")
  public void initialize(MediaContent mediaContent, final boolean muted) {
    controlsView.setVisibility(View.GONE);
    VideoController videoController = mediaContent.getVideoController();
    if (mediaContent != null
        && mediaContent.getHasVideoContent()
        && videoController != null
        && videoController.isCustomControlsEnabled()) {
      controlsView.setVisibility(View.VISIBLE);
      onVideoMute(muted);
      muteButton.setOnClickListener(unusedView -> videoController.mute(!videoController.isMuted()));
      playButton.setOnClickListener(
          unusedView -> {
            if (isVideoPlaying) {
              videoController.pause();
            } else {
              videoController.play();
            }
          });
    }
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onVideoMute(final boolean muted) {
    Drawable img;
    if (muted) {
      img = ContextCompat.getDrawable(getContext(), R.drawable.video_mute);
    } else {
      img = ContextCompat.getDrawable(getContext(), R.drawable.video_unmute);
    }
    muteButton.setImageDrawable(img);
    }

  @SuppressLint("SetTextI18n")
  @Override
  public void onVideoPause() {
    Drawable img = ContextCompat.getDrawable(getContext(), R.drawable.video_play);
    playButton.setImageDrawable(img);
    isVideoPlaying = false;
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onVideoPlay() {
    Drawable img = ContextCompat.getDrawable(getContext(), R.drawable.video_pause);
    playButton.setImageDrawable(img);
    isVideoPlaying = true;
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onVideoStart() {
    Drawable img = ContextCompat.getDrawable(getContext(), R.drawable.video_pause);
    playButton.setImageDrawable(img);
    isVideoPlaying = true;
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onVideoEnd() {
    Drawable img = ContextCompat.getDrawable(getContext(), R.drawable.video_play);
    playButton.setImageDrawable(img);
    isVideoPlaying = false;
  }
}

File: gma-next-gen-sdk-android-examples-main/kotlin/nextgenexample/app/src/main/java/com/example/next_gen_example/native/CustomVideoControlsView.kt
package com.example.next_gen_example.native

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageButton
import android.widget.LinearLayout
import com.example.next_gen_example.R
import com.google.android.libraries.ads.mobile.sdk.common.VideoController
import com.google.android.libraries.ads.mobile.sdk.nativead.MediaContent

/**
 * This view represents the status of a video controller and also displays custom controls for the
 * video controller when appropriate.
 */
class CustomVideoControlsView
@JvmOverloads
constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) :
  LinearLayout(context, attrs, defStyle), VideoController.VideoLifecycleCallbacks {
  private val playButton: ImageButton
  private val muteButton: ImageButton
  private val controlsView: View
  private var isVideoPlaying: Boolean = true

  init {
    val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    inflater.inflate(R.layout.custom_video_controls, this, true)
    playButton = findViewById(R.id.btn_play)
    muteButton = findViewById(R.id.btn_mute)
    controlsView = findViewById(R.id.video_controls)
    controlsView.visibility = View.GONE
  }

  fun destroy() {
    controlsView.visibility = View.GONE
  }

  /*
   * Sets up the custom controls view with the provided VideoController
   * and mute state, video lifecycle callbacks.
   */
  fun initialize(mediaContent: MediaContent?, muted: Boolean) {
    controlsView.visibility = View.GONE
    val videoController = mediaContent?.videoController
    if (
      mediaContent != null && videoController != null && videoController.isCustomControlsEnabled
    ) {
      controlsView.visibility = View.VISIBLE
      onVideoMute(muted)
      muteButton.setOnClickListener { videoController.mute(!videoController.isMuted) }
      playButton.setOnClickListener {
        if (isVideoPlaying) {
          videoController.pause()
        } else {
          videoController.play()
        }
      }
    }
  }

  override fun onVideoMute(muted: Boolean) {
    val muteResource = if (muted) R.drawable.video_mute else R.drawable.video_unmute
    muteButton.setImageResource(muteResource)
    super.onVideoMute(muted)
  }

  override fun onVideoPause() {
    playButton.setImageResource(R.drawable.video_play)
    isVideoPlaying = false
    super.onVideoPause()
  }

  override fun onVideoPlay() {
    playButton.setImageResource(R.drawable.video_pause)
    isVideoPlaying = true
    super.onVideoPlay()
  }

  override fun onVideoStart() {
    playButton.setImageResource(R.drawable.video_pause)
    isVideoPlaying = true
    super.onVideoStart()
  }

  override fun onVideoEnd() {
    playButton.setImageResource(R.drawable.video_play)
    isVideoPlaying = false
    super.onVideoEnd()
  }
}
